pkg/podman/podman.go

```diff
@@ -89,6 +89,33 @@ func (o *PodmanCli) PlayKube(pod *corev1.Pod) error {
 	return nil
 }
 
+func (o *PodmanCli) KubeGenerate(name string) (*corev1.Pod, error) {
+	serializer := jsonserializer.NewSerializerWithOptions(
+		jsonserializer.SimpleMetaFactory{},
+		scheme.Scheme,
+		scheme.Scheme,
+		jsonserializer.SerializerOptions{
+			Yaml: true,
+		},
+	)
+
+	cmd := exec.Command(o.podmanCmd, "kube", "generate", name)
```

Comment:
I am running into the following error while running `ODO_EXPERIMENTAL_MODE=t odo delete component`, it seems it has something to do with this command:
```sh
odo delete component                                                                                                         
============================================================================                                                            
⚠ Experimental mode enabled. Use at your own risk.
More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
============================================================================

Searching resources to delete, please wait...
 ⚠  failed to get deployment "my-go-app-app": Get "https://192.168.49.2:8443/apis/apps/v1/namespaces/default/deployments/my-go-app-app": dial tcp 192.168.49.2:8443: i/o timeout
 ✗  exit status 125: Error: unrecognized command `podman kube`
Try 'podman --help' for more information.

```

But when I tried to use `podman play kube ...`, I ran into another error: " ✗  exit status 125: Error: accepts 1 arg(s), received 2"

Reply:
Which version of podman are you using?

Reply:
```sh
$ podman version
Version:      3.4.7
API Version:  3.4.7
Go Version:   go1.16.15
Built:        Mon Jul 11 04:05:25 2022
OS/Arch:      linux/amd64
```

Reply:
Is there a minimum version of podman that we need?

Reply:
The devs are done based on podman v4 (v4.3.0 exactly)

Reply:
Okay. There's one more error (might be out of context for this PR) that I am facing while using `odo dev`.

I am not connected to any cluster atm, I usually use minikube, which is in Stopped state.
When I run `ODO_EXPERIMENTAL_MODE=t odo dev --run-on=podman`, it panics because of https://github.com/redhat-developer/odo/blob/0c008c6be9666bcd61d6d3874fb6b60fe5b6f917/pkg/component/component.go#L94 because `GetRunningPodFromSelector` has not been implemented yet for podmanClient.

Are you able to reproduce this as well?

Reply:
Yes, this is something that needs to be done. I just created an issue to follow it: https://github.com/redhat-developer/odo/issues/6434